File: tuan02/bai42/bai42.py
def cua_so_truot(a: list[int], k: int) -> tuple[int, int]:
    n = len(a)
    if n < k or k <= 0:
        return (0, -1)

    current_sum = sum(a[:k])
    max_sum = current_sum
    best_start = 0

    for i in range(k, n):
        current_sum += a[i] - a[i - k]
        if current_sum > max_sum:
            max_sum = current_sum
            best_start = i - k + 1

    return (max_sum, best_start)

if __name__ == "__main__":
    import sys
    tokens = sys.stdin.read().split()
    if tokens:
        n = int(tokens[0])
        k = int(tokens[1])
        idx = 2
        if idx < len(tokens) and tokens[idx] == "/":
            idx += 1
        a = [int(x) for x in tokens[idx:idx+n]]
        res = cua_so_truot(a, k)
        print(f"{res[0]} {res[1]}")

File: tuan02/bai42/bai42.jl
function cua_so_truot(a::Vector{Int64}, k::Int64)::Tuple{Int64, Int64}
    n = length(a)
    if n < k || k <= 0
        return (0, -1)
    end

    current_sum = sum(a[1:k])
    max_sum = current_sum
    best_start = 1

    for i in (k + 1):n
        current_sum += a[i] - a[i - k]
        if current_sum > max_sum
            max_sum = current_sum
            best_start = i - k + 1
        end
    end

    # Tra ve 0-based index cho dong bo voi C++ va Python
    return (max_sum, best_start - 1)
end

input_str = read(stdin, String)
tokens = split(input_str)
if !isempty(tokens)
    n = parse(Int64, tokens[1])
    k = parse(Int64, tokens[2])
    idx = 3
    if idx <= length(tokens) && tokens[idx] == "/"
        idx += 1
    end
    a = [parse(Int64, tokens[i]) for i in idx:(idx+n-1)]
    res = cua_so_truot(a, k)
    println("$(res[1]) $(res[2])")
end